[bug]: Asterisk not building C++ source due to "template" in ast_file_mkftemp

### Severity

Minor

### Versions

Asterisk 18.10.0

### Components/Modules

File.h

### Operating Environment

Ubuntu 22.04.4 LTS
gcc version 11.4.0


### Frequency of Occurrence

None

### Issue Description

I am trying to build Asterisk 18.10.0 however I am met with:

/usr/include/asterisk/file.h:148:30: error: expected ',' or '...' before 'template'
  148 | FILE *ast_file_mkftemp(char *template, mode_t mode);

I cannot find any other posts related to this issue, so wondering if anyone has any pointers?

### Relevant log output

```shell
[ 73%] Building CXX object phones_status/tests/CMakeFiles/NewAsterisk_Main_run.dir/app_phonestatus_tests.cxx.o
In file included from /source/AsteriskApps/phones_status/tests/../app_phonestatus.c:21,
                 from /source/AsteriskApps/phones_status/tests/app_phonestatus_tests.cxx:5:
/usr/include/asterisk/file.h:148:30: error: expected ',' or '...' before 'template'
  148 | FILE *ast_file_mkftemp(char *template, mode_t mode);
```


### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

--- a/include/asterisk/file.h
+++ b/include/asterisk/file.h
@@ -140,12 +140,13 @@ int ast_filecopy(const char *oldname, const char *newname, const char *fmt);
 
 /*!
  * \brief same as mkstemp, but return a FILE
- * \param template The template for the unique file name to generate. Modified in place to return the file name.
+ * \param template_name The template for the unique file name to generate.
+ *                      Modified in place to return the file name.
  * \param mode The mode for file permissions
  *
  * \return FILE handle to the temporary file on success or NULL if creation failed
  */
-FILE *ast_file_mkftemp(char *template, mode_t mode);
+FILE *ast_file_mkftemp(char *template_name, mode_t mode);
 
 /*!
  * \brief Create a temporary file located at path

--- a/main/file.c
+++ b/main/file.c
@@ -184,11 +184,11 @@ int ast_format_def_unregister(const char *name)
 	return res;
 }
 
-FILE *ast_file_mkftemp(char *template, mode_t mode)
+FILE *ast_file_mkftemp(char *template_name, mode_t mode)
 {
 	FILE *p = NULL;
-	int pfd = mkstemp(template);
-	chmod(template, mode);
+	int pfd = mkstemp(template_name);
+	chmod(template_name, mode);
 	if (pfd > -1) {
 		p = fdopen(pfd, "w+");
 		if (!p) {

